go/libraries/doltcore/sqle/alterschema.go

```diff
@@ -778,6 +779,67 @@ func dropColumn(ctx context.Context, tbl *doltdb.Table, colName string) (*doltdb
 	return tbl.UpdateSchema(ctx, newSch)
 }
 
+// backupFkcIndexesForKeyDrop finds backup indexes to cover foreign key references during a primary
+// key drop. If multiple indexes are valid, we sort by unique and select the first.
+// This will not work with a non-pk index drop without an additional index filter argument.
+func backupFkcIndexesForPkDrop(ctx *sql.Context, sch schema.Schema, fkc *doltdb.ForeignKeyCollection) ([]doltdb.FkIndexUpdate, error) {
```

Comment:
After some more thought, I think this kind of backup-searching functionality belongs in GMS on the PK drop path. We have too much stuff in Dolt that should be in GMS, and we can unintentionally break things because of it. If we later adjust GMS and introduce a new interface or something dealing with PK alteration, then it's not going to be immediately clear that we're also affecting foreign keys (and also that foreign keys should even be considered in the first place). This is very similar to the issues with the recent foreign key breakage in the new format as well (from the `RewriteInserter` addition, Dolt just does too much and GMS too little).